snap/hooks/configure

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+FEATURE_GATES="$(snapctl get feature-gates)"
+
+if [ -z "$FEATURE_GATES" ]
+then
+      echo "no feature gates configured"
+      echo -n "" > $SNAP_DATA/feature-gates.conf
+else
+    # Validate the feature-gates using the otelcol validate command
+    if ! $SNAP/bin/otelcol validate  --feature-gates=$FEATURE_GATES --config="$SNAP/etc/config.yaml" > /dev/null 2>&1; then
+        echo "\"$FEATURE_GATES\" is not a valid feature gate definition" >&2
```

Comment:
Iiuc, the `validate` here validates the config file, not the cli args.
If so, then the configure hook is probably the wrong place for this validation.

Is there a list of feature gates otelcol can output and the hook to compare against?

Reply:
it validates the config within the context of the cli args (and the cli args themselves)
So it will fail if:
- the argument is invalid e.g. the feature gate doesn't exist or is misspelled
- the config references/uses a gated feature but the feature isn't enabled by a matching flag.